src/storage/data_directory.rs

```diff
@@ -0,0 +1,283 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use parity_codec::Codec;
+use parity_codec_derive::{Encode, Decode};
+use srml_support::{StorageMap, decl_module, decl_storage, decl_event, ensure, Parameter, dispatch};
+use runtime_primitives::traits::{Member, MaybeSerializeDebug, MaybeDebug};
+use system::{self, ensure_signed};
+use crate::traits::{IsActiveMember, IsActiveDataObjectType, ContentIdExists};
+use crate::storage::data_object_type_registry::Trait as DOTRTrait;
+
+pub trait Trait: timestamp::Trait + system::Trait + DOTRTrait + MaybeDebug {
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    type ContentId: Parameter + Member + Codec + Default + Clone
+        + MaybeSerializeDebug + PartialEq;
+
+    type IsActiveMember: IsActiveMember<Self>;
+    type IsActiveDataObjectType: IsActiveDataObjectType<Self>;
+}
+
+static MSG_DUPLICATE_CID: &str = "Content with this ID already exists!";
+static MSG_CID_NOT_FOUND: &str = "Content with this ID not found!";
+static MSG_LIAISON_REQUIRED: &str = "Only the liaison for the content may modify its status!";
+static MSG_CREATOR_MUST_BE_MEMBER: &str = "Only active members may create content!";
+static MSG_DO_TYPE_MUST_BE_ACTIVE: &str = "Cannot create content for inactive of missing data object type!";
+
+#[derive(Clone, Encode, Decode, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum LiaisonJudgement
+{
+    Pending,
+    Rejected,
+    Accepted,
+}
+
+impl Default for LiaisonJudgement {
+    fn default() -> Self {
+        LiaisonJudgement::Pending
+    }
+}
+
+#[derive(Clone, Encode, Decode, PartialEq)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub struct DataObject<T: Trait> {
+    pub data_object_type: <T as DOTRTrait>::DataObjectTypeId,
+    pub size: u64,
+    pub added_at_block: T::BlockNumber,
+    pub added_at_time: T::Moment,
+    pub origin: T::AccountId,
+    pub liaison: T::AccountId,
+    pub liaison_judgement: LiaisonJudgement,
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as DataDirectory {
+        // Mapping of Content ID to Data Object
+        pub ContentMap get(content): map T::ContentId => Option<DataObject<T>>;
+    }
+}
+
+decl_event! {
+    pub enum Event<T> where
+        <T as Trait>::ContentId,
+        <T as system::Trait>::AccountId
+    {
+        // The account is the Liaison that was selected
+        ContentAdded(ContentId, AccountId),
+        ContentAccepted(ContentId),
```

Comment:
We could do that. For `ContentAdded` it's pretty important, because uploaders can then skip trying to query the record for finding the liaison. For the two state changes, the uploader has already been in contact with the liaison.

Either works for me.